File: algorithms/[7]整数反转/Solution.java
class Solution {
    public int reverse(int x) {
        long n = 0;
        while (x != 0) {
            n = n * 10 + x % 10;
            x = x / 10;
        }
        return (int)n == n ? (int)n : 0;
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(s.reverse(123));
        System.out.println(s.reverse(-123));
        System.out.println(s.reverse(120));
    }
}

File: algorithms/[7]整数反转/Solution.cs
using System;

public class Solution
{
    /// <summary>
    /// Math
    /// time O(log|x|), space O(1), log|x| 即 x 十进制的位数
    /// </summary>
    public int Reverse(int x)
    {
        int rev = 0;

        while (x != 0)
        {
            if (rev < int.MinValue / 10 || rev > int.MaxValue / 10)
            {
                return 0;
            }
            rev = rev * 10 + x % 10;
            x /= 10;
        }

        return rev;
    }
}